app/models/nic/interface.rb

```diff
@@ -23,7 +23,10 @@ class Interface < Base
     delegate :network, :to => :subnet
 
     def vlanid
-      self.tag.blank? ? self.subnet.vlanid : self.tag
+      return self.tag unless self.tag.blank?
+      return self.subnet.vlanid if self.subnet
```

Comment:
@ares while your reasoning is sound code-wise, the `subnet` and `subnet6` _should_ have the same vlanid (if the subnets are present); after all, this is the same connection, only different protocol.

The case I'm solving here, is the one where you don't have a IPv4 `subnet` at all.

With that in mind, should I or should I not check for blank? :-)

Having a blank vlanid for the v4 subnet and non-blank vlanid for v6 subnet is rather a user error in my opinion?